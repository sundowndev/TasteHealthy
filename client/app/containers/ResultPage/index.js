/* eslint-disable jsx-a11y/anchor-is-valid */
/*
 * ResultPage
 *
 * This is the result page, at the '/result' route
 *
 */

import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import '../../styles/ResultPage.css';
import axios from 'axios';
import { map, assoc, prop } from 'ramda';

// Components
import SidebarComponent from './components/SideBarComponent';
import NutriscoreComponent from './components/NutriscoreComponent';
import OriginComponent from './components/OriginComponent';
import PackagingComponent from './components/PackagingComponent';
import ProductComponent from './components/ProductComponent';
import SubstituteProductComponent from './components/SubstituteProductComponent';

// Food data
import {
  getMealsNutriScore,
  getMealsOrigin,
  getMealsPackaging,
  getCalories,
  getSodium,
  getSalt,
  getProteins,
  getFat,
  getSugar,
  getAdditives,
} from './utils/getFoodData';

import progressBar from './components/ProgressBarComponent';

type Props = {
  match: {
    params: {
      mealType: string,
    },
  },
  mealsData: any,
};

const ResultPage = (props: Props) => {
  const { mealType } = props.match.params;
  const { mealsData } = props;
  const [mealsElements, setMealsElements] = useState(
    mealType === 'total'
      ? [].concat.apply(
        [],
        [
          props.mealsData.breakfast.consummedAliments,
          props.mealsData.lunch.consummedAliments,
          props.mealsData.snack.consummedAliments,
          props.mealsData.dinner.consummedAliments,
        ],
      )
      : props.mealsData[mealType].consummedAliments,
  );
  const [usedMealsElements, setUsedMealsElements] = useState(mealsElements);
  const [arrayIndex, setArrayIndex] = useState(
    new Array(mealsElements.length).fill(true),
  );

  const [substitute, setSubstitute] = useState([]);

  const updateMeals = () => {
    window.scrollTo(0, 0);
    setTimeout(() => {
      const newMealsElements = mealsElements.map((el, index) => {
        if (!arrayIndex[index]) {
          return substitute[index];
        }
        return el;
      });
      setUsedMealsElements(newMealsElements);
    }, 500);
  };

  useEffect(() => {
    const mealsElements2 =
      mealType === 'total'
        ? [].concat.apply(
          [],
          [
            props.mealsData.breakfast.consummedAliments,
              props.mealsData.lunch.consummedAliments,
            props.mealsData.snack.consummedAliments,
            props.mealsData.dinner.consummedAliments,
            ],
        )
        : props.mealsData[mealType].consummedAliments;
    const promises = [];
    for (let i = 0; i < mealsElements2.length; i++) {
      promises.push(
        axios
          .get(
            `http://localhost:3000/categories/${
              mealsElements2[i].categoryId
            }/products`,
          )
          .then(data => data.data.items[0]),
      );
    }
    setMealsElements(mealsElements2);
    setUsedMealsElements(mealsElements2);
    // setArrayIndex(new Array(mealsElements.length).fill(true));

    Promise.all(promises)
      .then(rr => {
        setSubstitute(map(assoc('quantity', 100), rr));
      })
      .catch(err => {
        console.log(err);
      });
  }, [mealType]);

  const toggleCheck = index => {
    const newArray = arrayIndex.map((el, indexEl) =>
      index === indexEl ? !el : el,
    );
    setArrayIndex(newArray);
  };

  const checkReRender = () => {
    const array1 = map(prop('id'), mealsElements);
    const array2 = map(prop('id'), usedMealsElements);
    const quantity1 = map(prop('quantity'), mealsElements);
    const quantity2 = map(prop('quantity'), usedMealsElements);
    if (
      JSON.stringify(quantity1) !== JSON.stringify(quantity2) ||
      JSON.stringify(array1) !== JSON.stringify(array2)
    ) {
      return true;
    }
    return false;
  };

  const getMeal = () => {
    switch (mealType) {
      case 'breakfast':
        return 'Petit Déjeuner';
      case 'lunch':
        return 'Déjeuner';
      case 'snack':
        return 'Goûter';
      case 'dinner':
        return 'Dîner';
      default:
        return 'Total';
    }
  };

  return (
    <div className="app">
      <div className="app__top" />
      <div className="app__right" />
      <div className="app__bottom" />
      <div className="app__left" />

      <div className="app__logo" />

      <div className="app__container">
        <SidebarComponent
          mealType={mealType}
          props={props}
          mealsData={mealsData}
        />

        <div className="app__content">
          <p className="app__content__title">
            Produits consommés - {getMeal()}
          </p>
          <img
            className="app__content__block__svg__half-circle"
            src={require('../../images/1.svg')}
          />
          <img
            className="app__content__block__svg__rectangle"
            src={require('../../images/2.svg')}
          />
          <div className="app__content__block__flex">
            <div
              className="app__content__block__flex__left"
              style={{ alignItems: 'baseline' }}
            >
              <div className="app__content__block__flex__component">
                {progressBar(
                  getSugar(usedMealsElements),
                  70,
                  70,
                  16,
                  'g',
                  checkReRender() ? getSugar(mealsElements) : null,
                )}
                <p className="app__content__block__flex__left__detail">Sucre</p>
              </div>
              <div className="app__content__block__flex__component">
                {progressBar(
                  getSodium(usedMealsElements),
                  70,
                  70,
                  16,
                  'g',
                  checkReRender() ? getSodium(mealsElements) : null,
                )}
                <p className="app__content__block__flex__left__detail">
                  Sodium
                </p>
              </div>
              <div className="app__content__block__flex__component">
                {progressBar(
                  getSalt(usedMealsElements),
                  70,
                  70,
                  16,
                  'g',
                  checkReRender() ? getSalt(mealsElements) : null,
                )}
                <p className="app__content__block__flex__left__detail">Sel</p>
              </div>
            </div>
            <div className="app__content__block__flex__middle">
              <div>
                {progressBar(
                  getCalories(usedMealsElements),
                  280,
                  280,
                  35,
                  'cal',
                  checkReRender() ? getCalories(mealsElements) : null,
                  props.caloriesData.calories,
                )}
              </div>
              <p
                style={{
                  fontFamily: 'Apercu',
                  marginTop: '10px',
                  textAlign: 'center',
                  fontSize: '15px',
                }}
              >
                | <br />
                <br /> Consommation recommandée
              </p>
            </div>

            <div
              style={{ alignItems: 'flex-end' }}
              className="app__content__block__flex__right"
            >
              <div className="app__content__block__flex__component">
                <p className="app__content__block__flex__right__detail">
                  Protéines
                </p>
                {progressBar(
                  getProteins(usedMealsElements),
                  70,
                  70,
                  16,
                  'g',
                  checkReRender() ? getProteins(mealsElements) : null,
                )}
              </div>
              <div className="app__content__block__flex__component">
                <p className="app__content__block__flex__right__detail">
                  Graisses
                </p>
                {progressBar(
                  getFat(usedMealsElements),
                  70,
                  70,
                  16,
                  'g',
                  checkReRender() ? getFat(mealsElements) : null,
                )}
              </div>
              <div className="app__content__block__flex__component">
                <p className="app__content__block__flex__right__detail">
                  Additifs
                </p>
                {progressBar(
                  getAdditives(usedMealsElements),
                  70,
                  70,
                  16,
                  '',
                  checkReRender() ? getAdditives(mealsElements) : null,
                )}
              </div>
            </div>
          </div>

          <div className="app__content__blocks app__content__blocks--three">
            <NutriscoreComponent
              letter={getMealsNutriScore(usedMealsElements)}
            />

            <OriginComponent origin={getMealsOrigin(usedMealsElements)} />

            <PackagingComponent data={getMealsPackaging(usedMealsElements)} />
          </div>

          <div className="app__content__substitute-products">
            <p className="app__content__title">Produits de substitutions</p>
            <div className="app__content__block">
              <img
                src={require('../../images/octo.svg')}
                className="app__content__block__svg__special-rectangle"
              />
              <div className="app__content__blocks">
                <div className="app__content__substitute-products__content">
                  <p className="app__content__substitute-products__content__title">
                    Mes produits
                  </p>
                  {mealsElements.map((meal, index) => (
                    <ProductComponent
                      key={index}
                      mealProps={meal}
                      isChecked={arrayIndex[index]}
                      onClick={() =>
                        !arrayIndex[index] ? toggleCheck(index) : null
                      }
                    />
                  ))}
                </div>

                <div className="app__content__substitute-products__content app__content__substitute-products__content--reverse">
                  <p className="app__content__substitute-products__content__title">
                    Les produits conseillés
                  </p>
                  {substitute.map((meal, index) => (
                    <SubstituteProductComponent
                      key={index}
                      mealProps={meal}
                      isChecked={!arrayIndex[index]}
                      onClick={() =>
                        arrayIndex[index] ? toggleCheck(index) : null
                      }
                    />
                  ))}
                </div>
              </div>
              <button
                onClick={() => updateMeals()}
                className="app__content__block__button"
              >
                Visualiser avec les nouveaux produits
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

function mapStateToProps(state) {
  const { meals, calories } = state;
  return { mealsData: meals, caloriesData: calories };
}

export default connect(mapStateToProps)(ResultPage);
