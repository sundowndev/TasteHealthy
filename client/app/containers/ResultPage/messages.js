/*
 * HomePage Messages
 *
 * This contains all the text for the ResultPage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.ResultPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ResultPage container!',
  },
});
