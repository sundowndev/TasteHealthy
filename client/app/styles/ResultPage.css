@font-face {
  font-family: "Apercu";
  src: url("fonts/Apercu-Bold.eot");
  src: url("fonts/Apercu-Bold.eot?#iefix") format("embedded-opentype"), url("fonts/Apercu-Bold.woff") format("woff"), url("fonts/Apercu-Bold.ttf") format("truetype"), url("fonts/Apercu-Bold.svg#Apercu") format("svg");
  font-weight: 700;
  font-style: normal; }

@font-face {
  font-family: "Apercu";
  src: url("fonts/Apercu-Light.eot");
  src: url("fonts/Apercu-Light.eot?#iefix") format("embedded-opentype"), url("fonts/Apercu-Light.woff") format("woff"), url("fonts/Apercu-Light.ttf") format("truetype"), url("fonts/Apercu-Light.svg#Apercu") format("svg");
  font-weight: 300;
  font-style: normal; }

@font-face {
  font-family: "Apercu";
  src: url("fonts/Apercu-Regular.eot");
  src: url("fonts/Apercu-Regular.eot?#iefix") format("embedded-opentype"), url("fonts/Apercu-Regular.woff") format("woff"), url("fonts/Apercu-Regular.ttf") format("truetype"), url("fonts/Apercu-Regular.svg#Apercu") format("svg");
  font-weight: normal;
  font-style: normal; }

/* http://meyerweb.com/eric/tools/css/reset/ 
     v2.0 | 20110126
     License: none (public domain)
  */
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed,
figure, figcaption, footer, header, hgroup,
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline; }

/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure,
footer, header, hgroup, menu, nav, section {
  display: block; }

body {
  line-height: 1; }

ol, ul {
  list-style: none; }

blockquote, q {
  quotes: none; }

blockquote:before, blockquote:after,
q:before, q:after {
  content: '';
  content: none; }

table {
  border-collapse: collapse;
  border-spacing: 0; }

body {
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 100; }

*,
*::before,
*::after {
  box-sizing: border-box; }

.app {
  padding: calc(50px + 40px) calc(50px + 35px);
  background-color: #FBFBFB;
  z-index: -1;
  position: relative; }
  .app__top {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    background-color: #FFF;
    height: 50px;
    width: 100%; }
  .app__right {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    background-color: #FFF;
    height: 100%;
    width: 50px; }
  .app__bottom {
    position: absolute;
    right: 0;
    left: 0;
    bottom: 0;
    background-color: #FFF;
    height: 50px;
    width: 100%; }
  .app__left {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    background-color: #FFF;
    height: 100%;
    width: 50px; }
  .app__logo {
    position: absolute;
    top: 0;
    left: 0; }
  .app__container {
    padding-top: 30px;
    display: flex; }
    .app__container > *:first-child {
      flex: 15%; }
    .app__container > *:last-child {
      flex: 85%; }
  .app__sidebar {
    margin-top: 35px;
    text-align: center;
    margin-right: 10px; }
    .app__sidebar__title {
      text-align: center;
      font-family: 'Apercu', sans-serif;
      color: #33176E;
      font-size: 18px;
      letter-spacing: 0;
      font-weight: 300;
      margin-top: 20px; }
    .app__sidebar__content {
      margin-top: 40px; }
      .app__sidebar__content__block {
        text-align: left;
        background-color: #FFF;
        border-radius: 10px;
        border: 2px solid #F6F6F6;
        padding: 13px 10px;
        display: flex;
        align-items: center;
        justify-content: space-between; }
        .app__sidebar__content__block:hover {
          cursor: pointer; }
        .app__sidebar__content__block:not(:first-child) {
          margin-top: 15px; }
        .app__sidebar__content__block p {
          font-family: 'Apercu', sans-serif;
          color: #33176E;
          font-size: 15px;
          letter-spacing: 0;
          font-weight: 700; }
        .app__sidebar__content__block__add {
          margin-bottom: -3px; }
        .app__sidebar__content__block__checked {
          margin-bottom: -3px;
          margin-left: 2px; }
        .app__sidebar__content__block__edit {
          margin-bottom: -1px; }
  .app__content {
    margin-left: 10px;
    margin-top: -20px; }
    .app__content__title {
      font-family: 'Apercu', sans-serif;
      color: #928AA4;
      font-size: 35px;
      letter-spacing: 0;
      font-weight: 700;
      margin-bottom: 15px; }
      .app__content__title--md {
        font-size: 20px;
        margin-bottom: 10px; }
    .app__content__block {
      background-color: #FFF;
      padding: 50px 35px;
      border-radius: 5px;
      box-shadow: 0 6px 13px 0 rgba(21, 33, 56, 0.05);
      position: relative; }
      .app__content__block__svg__rectangle {
        position: absolute;
        top: -25px;
        right: -20px;
        z-index: -1; }
      .app__content__block__svg__special-rectangle {
        position: absolute;
        bottom: 0;
        left: 0;
        transform: translate(-40%, 40%);
        z-index: -1; }
      .app__content__block__svg__half-circle {
        top: -35px;
        left: 40%;
        transform: translateX(-50%);
        position: absolute;
        z-index: -1; }
      .app__content__block__svg__oval {
        position: absolute;
        z-index: -1; }
        .app__content__block__svg__oval--left {
          left: 0;
          bottom: 30px;
          transform: translateX(-50%); }
        .app__content__block__svg__oval--right {
          right: -30px;
          bottom: 70px; }
    .app__content__blocks {
      display: grid;
      margin-top: 70px;
      position: relative; }
      .app__content__blocks--three {
        grid-template-columns: repeat(3, 1fr);
        grid-column-gap: 30px; }
      .app__content__blocks .app__content__block {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        height: 100%; }
      .app__content__blocks .app__content__title {
        position: absolute;
        top: -30px; }
    .app__content .app__content__block__flex {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      align-items: center;
      background-color: #FFF;
      padding: 50px 35px;
      border-radius: 5px;
      box-shadow: 0 6px 13px 0 rgba(21, 33, 56, 0.05);
      position: relative; }
    .app__content .app__content__block__flex__left {
      display: flex;
      flex-direction: column; }
    .app__content .app__content__block__flex__component {
      display: flex;
      flex-direction: row;
      align-items: center; }
    .app__content .app__content__block__flex__right {
      display: flex;
      flex-direction: column; }
    .app__content .app__content__block__text__origin {
      text-transform: uppercase;
      font-family: 'Apercu', sans-serif;
      color: #33176E;
      font-size: 25px;
      letter-spacing: 0;
      font-weight: 700;
      text-align: center;
      margin-right: 20px; }
    .app__content__block--packaging__circle {
      width: 12px;
      height: 12px;
      border-radius: 100%;
      margin-right: 10px; }
    .app__content__block--packaging__content {
      display: flex;
      justify-content: space-between;
      width: 100%;
      margin-top: auto; }
      .app__content__block--packaging__content__column > div {
        display: flex;
        align-items: center; }
      .app__content__block--packaging__content__column:first-child > div:first-child .app__content__block--packaging__circle {
        background-color: #F2F5F7; }
      .app__content__block--packaging__content__column:first-child > div:nth-child(2) .app__content__block--packaging__circle {
        background-color: #C2D1D9; }
      .app__content__block--packaging__content__column:nth-child(2) > div:first-child .app__content__block--packaging__circle {
        background-color: #424B5A; }
      .app__content__block--packaging__content__column:nth-child(2) > div:nth-child(2) .app__content__block--packaging__circle {
        background-color: #A8C8D8; }
      .app__content__block--packaging__content__column > div:not(:first-child) {
        margin-top: 10px; }
      .app__content__block--packaging__content__column p {
        color: #33176E;
        font-family: 'Apercu', sans-serif;
        font-weight: 300;
        letter-spacing: 0;
        font-size: 15px; }
        .app__content__block--packaging__content__column p strong {
          font-weight: 700; }
    .app__content__substitute-products {
      margin-top: 35px; }
      .app__content__substitute-products .app__content__blocks {
        grid-template-columns: 1fr 1fr;
        grid-column-gap: 45px;
        margin-top: 0; }
      .app__content__substitute-products .app__content__substitute-products__content__title {
        text-transform: uppercase;
        color: #928AA4;
        font-size: 20px;
        font-weight: 400;
        font-family: 'Apercu', sans-serif;
        margin-bottom: 20px; }
      .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block {
        display: flex;
        align-items: center; }
        .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block:hover {
          cursor: pointer; }
        .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block:not(:first-child) {
          margin-top: 10px; }
        .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block > div:first-child {
          background-color: #FFF;
          border-radius: 5px;
          padding: 8px 15px;
          background: #FFFFFF;
          box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.12);
          width: 100%; }
          .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block > div:first-child strong {
            font-weight: 700;
            font-size: 15px;
            margin-bottom: 5px;
            display: block; }
          .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block > div:first-child p {
            font-size: 13px;
            font-weight: 300; }
          .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block > div:first-child p,
          .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block > div:first-child strong {
            font-family: 'Apercu', sans-serif;
            color: #33176E;
            letter-spacing: 0; }
        .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block__check {
          margin-left: 20px; }
          .app__content__substitute-products__content:not(.app__content__substitute-products__content--reverse) .app__content__substitute-products__block__check:not(.is-checked) {
            width: 20px;
            height: 20px;
            border-radius: 100%;
            border: 3px solid #DDDDDD; }
      .app__content__substitute-products__content--reverse .app__content__substitute-products__content__title {
        margin-left: 40px; }
      .app__content__substitute-products__content--reverse .app__content__substitute-products__block {
        display: flex;
        align-items: center; }
        .app__content__substitute-products__content--reverse .app__content__substitute-products__block:hover {
          cursor: pointer; }
        .app__content__substitute-products__content--reverse .app__content__substitute-products__block:not(:first-child) {
          margin-top: 10px; }
        .app__content__substitute-products__content--reverse .app__content__substitute-products__block > div:last-child {
          background-color: #FFF;
          border-radius: 5px;
          padding: 8px 15px;
          background: #FFFFFF;
          box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.12);
          width: 100%; }
          .app__content__substitute-products__content--reverse .app__content__substitute-products__block > div:last-child strong {
            font-weight: 700;
            font-size: 15px;
            margin-bottom: 5px;
            display: block; }
          .app__content__substitute-products__content--reverse .app__content__substitute-products__block > div:last-child p {
            font-size: 13px;
            font-weight: 300; }
          .app__content__substitute-products__content--reverse .app__content__substitute-products__block > div:last-child p,
          .app__content__substitute-products__content--reverse .app__content__substitute-products__block > div:last-child strong {
            font-family: 'Apercu', sans-serif;
            color: #33176E;
            letter-spacing: 0; }
        .app__content__substitute-products__content--reverse .app__content__substitute-products__block__check {
          margin-right: 20px; }
          .app__content__substitute-products__content--reverse .app__content__substitute-products__block__check:not(.is-checked) {
            width: 20px;
            height: 20px;
            border-radius: 100%;
            border: 3px solid #DDDDDD; }
      .app__content__substitute-products__block__check {
        display: flex;
        align-items: center; }
    .app__content__block__button {
      background-color: #D9C5F4;
      border-radius: 10px;
      font-size: 13px;
      letter-spacing: 0;
      color: #33176E;
      font-family: 'Apercu', sans-serif;
      font-weight: 700;
      text-align: center;
      text-decoration: none;
      padding: 20px 30px;
      display: block;
      margin-top: 60px;
      max-width: 300px;
      margin-left: auto;
      margin-right: auto; }
