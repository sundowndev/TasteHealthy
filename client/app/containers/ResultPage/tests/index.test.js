import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import configureStore from 'redux-mock-store';

import ResultPage from '../index';

// create any initial state needed
const initialState = {
  meals: {
    breakfast: {
      consummedAliments: [
        JSON.parse(`{
      "id": 7,
      "product_name": "All Butter Fruity Flapjack Cookies",
      "generic_name": "Biscuits pur beurre avec de l'avoine, des raisins secs, de la noix de coco et des abricots déshydratés.",
      "quantity": 225,
      "quantity_unity": "g",
      "image_url": "https://static.openfoodfacts.org/images/products/00011709/front_fr.12.400.jpg",
      "origins": "unknown",
      "packaging": "sachet,plastique",
      "manufacturing_places": "Royaume-Uni",
      "countries": "France,Royaume-Uni",
      "labels": "Végétarien,Point Vert,Pur beurre",
      "purchase_places": "Royaume-Uni,France",
      "stores": "Marks & Spencer",
      "nutrition_facts": {
        "energy_100g": "1887",
        "fat_100g": "20.5",
        "saturated_fat_100g": "13.3",
        "carbohydrates_100g": "59.3",
        "sugars_100g": "36.5",
        "fiber_100g": "4.2",
        "proteins_100g": "5.1",
        "salt_100g": "0.779999999999999",
        "sodium_100g": "0.307086614173228",
        "fruits_vegetables_nuts_estimate_100g": "27",
        "nutrition_score_fr_100g": "22",
        "nutrition_score_uk_100g": "22"
      },
      "misc_data": {
        "serving_size_g": "23 g",
        "additives_n": 5,
        "additives": "[ sucre -> fr:sucre  ]  [ beurre -> fr:beurre  ]  [ lait -> fr:lait  ]  [ 15 -> fr:15  ]  [ flocons-d-avoine -> fr:flocons-d-avoine  ]  [ flocons-d -> fr:flocons-d  ]  [ flocons -> fr:flocons  ]  [ contient-du-gluten -> fr:contient-du-gluten  ]  [ contient-du -> fr:contient-du  ]  [ contient -> fr:contient  ]  [ farine-de-ble -> fr:farine-de-ble  ]  [ farine-de -> fr:farine-de  ]  [ farine -> fr:farine  ]  [ contient-du-gluten -> fr:contient-du-gluten  ]  [ contient-du -> fr:contient-du  ]  [ contient -> fr:contient  ]  [ avec-du-ble -> fr:avec-du-ble  ]  [ avec-du -> fr:avec-du  ]  [ avec -> fr:avec  ]  [ du-carbonate-de-calcium -> en:e170  -> exists  -- mandatory_additive_class: en:acidity-regulator, en:anti-caking-agent, en:stabiliser, en:firming-agent, en:flour-treatment-agent, en:glazing-agent, en:colour (current: ingredient)  ]  [ du-carbonate-de -> fr:du-carbonate-de  ]  [ du-carbonate -> en:e502  -> exists  -- ok  ]  [ du-fer -> fr:du-fer  ]  [ du -> fr:du  ]  [ de-la-niacine -> en:e375  -> exists  -- mandatory_additive_class: en:colour-retention-agent (current: ingredient)  ]  [ de-la -> fr:de-la  ]  [ de -> fr:de  ]  [ de-la-thiamine -> fr:de-la-thiamine  ]  [ de-la -> fr:de-la  ]  [ de -> fr:de  ]  [ raisins-secs -> fr:raisins-secs  ]  [ raisins -> fr:raisins  ]  [ 12 -> en:fd-c  ]  [ noix-de-coco-dessechee -> fr:noix-de-coco-dessechee  ]  [ noix-de-coco -> fr:noix-de-coco  ]  [ noix-de -> fr:noix-de  ]  [ noix -> fr:noix  ]  [ 8 -> en:fd-c  ]  [ abricots-seches -> fr:abricots-seches  ]  [ abricots -> fr:abricots  ]  [ 7 -> en:fd-c  ]  [ abricots -> fr:abricots  ]  [ farine-de-riz -> fr:farine-de-riz  ]  [ farine-de -> fr:farine-de  ]  [ farine -> fr:farine  ]  [ conservateur -> fr:conservateur  ]  [ e220 -> en:e220  -> exists  -- ok  ]  [ sulfites -> fr:sulfites  ]  [ golden-syrup -> fr:golden-syrup  ]  [ golden -> fr:golden  ]  [ sirop-de-sucre-inverti -> fr:sirop-de-sucre-inverti  ]  [ sirop-de-sucre -> fr:sirop-de-sucre  ]  [ sirop-de -> fr:sirop-de  ]  [ sirop -> fr:sirop  ]  [ oeufs-pasteurises -> fr:oeufs-pasteurises  ]  [ oeufs -> fr:oeufs  ]  [ agent-de-levage -> fr:agent-de-levage  ]  [ agent-de -> fr:agent-de  ]  [ agent -> fr:agent  ]  [ bicarbonate-de-sodium -> en:e500ii  -> exists  -- ok  ]  [ e450 -> en:e450  -> exists  -- ok  ]  [ e503 -> en:e503  -> exists  -- mandatory_additive_class: en:raising-agent (current: en:preservative)  -- e-number  ]  [ sel -> fr:sel  ]  [ melasse -> fr:melasse  ]",
        "ingredients_from_palm_oil_n": 0,
        "nutrition_grade_fr": "e",
        "nutrition_score_fr_100g": "22",
        "nutrition_score_uk_100g": "22",
        "ingredients_text": "Sucre • Beurre (_Lait_) (15%) • flocons d'_avoine_ (contient du _gluten_) • Farine de _blé_(contient du _gluten_) (avec du _blé_, du carbonate de calcium, du fer, de la niacine, de la thiamine) • Raisins secs (12%) • Noix de coco desséchée (8 %) • Abricots séchés (7 %) (Abricots • Farine de riz • Conservateur: E220 (_Sulfites_)) • Golden syrup (sirop de sucre inverti) • _œufs_ pasteurisés • Agent de levage: Bicarbonate de sodium, E450, E503, sel, Mélasse."
      },
      "createdAt": "2019-05-29T12:03:25.853Z",
      "updatedAt": "2019-05-29T12:03:25.853Z",
      "categoryId": 5
    }`),
      ],
    },
    lunch: {},
    snack: {},
    dinner: {},
  },
};

const mockStore = configureStore();

const match = {
  params: {
    mealType: 'breakfast',
  },
};

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <ResultPage
          blou="zefze"
          match={match}
          store={mockStore(initialState)}
        />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
