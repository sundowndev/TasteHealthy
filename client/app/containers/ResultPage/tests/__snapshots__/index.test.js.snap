// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the ResultPage container!
  </span>
</h1>
`;
